controllers/article-controller.js

```diff
@@ -26,4 +28,22 @@ const getAllArticles = (request, response, next) => {
     });
 };
 
-module.exports = { getArticleById, getAllArticles };
+const getAllCommentsByArticleId = (request, response, next) => {
+  const { article_id } = request.params;
+  const promises = [allCommentsForArticle(article_id)];
+
+  if (article_id) {
+    promises.push(checkArticleExists(article_id));
+  }
```

Comment:
This if statement here isn't necessary because the endpoint `/api/articles/**:article_id/**comments` will always have an article id, otherwise this endpoint won't be called.

So you can have this directly in the const above!